src/odemis/driver/scanner.py

```diff
@@ -112,3 +124,211 @@ def updateMetadata(self, md):
 
     def getMetadata(self):
         return self._external.getMetadata()
+
+    def claimBeam(self, claim):
+        """
+        Used to claim the beam by blanking the beam on the scanner of the XT client when the blanker
+        mode is set to automatic (blanker.value = None)
+        :param claim (boolean): True for unblanking and False for blanking
+        """
+        if self.blanker.value is None:
+            self._setBlanker(not claim)  # Set the blanker using the setter of the VA
+
+
+class CompositedDetector(model.Detector):
+    '''
+    A generic Detector which takes 2 dependencies to create a one detector. It's
+    essentially a wrapper to an Detector to generate data using the "external"
+    detector while manipulating the "internal" detector.
+    '''
+    def __init__(self, name, role, dependencies, **kwargs):
+        '''
+        dependencies (dict string->model.HwComponent): the dependencies
+            There must be exactly two dependencies "external" and "internal".
+        Raise:
+          ValueError: if the dependencies are not compatible
+        '''
+        # we will fill the set of dependencies with Components later in ._dependencies
+        model.Emitter.__init__(self, name, role, dependencies=dependencies, **kwargs)
+
+        # Check the dependencies
+        extnl = dependencies["external"]
+        if not isinstance(extnl, ComponentBase):
+            raise ValueError("Dependency external is not a component.")
+        if not hasattr(extnl, "data") and not hasattr(extnl, "shape"):
```

Comment:
You can make a single `if`, which check all 3 properties, and fails with "...is not a Detector component".
Also note that hasattr() doesn't really work with Pyro, because Pyro always claims there is something. So you'd need to check that they are present *and* they are of the right type (data -> DataFlowBase, shape -> tuple).

Reply:
So you mean like this:

>         if not isinstance(extnl, ComponentBase)\
>                 and not hasattr(extnl, "data") and isinstance(extnl.data, DataFlowBase)\
>                 and not hasattr(extnl, "shape") and isinstance(extnl.shape, tuple):
>                 raise ValueError("Dependency external is not a Detector component.")

Reply:
Approximately like that, but I think you need to tweak the logic to something like that:
```
if not isinstance(extnl, ComponentBase)\
            or not (hasattr(extnl, "data") and isinstance(extnl.data, DataFlowBase))\
            or not (hasattr(extnl, "shape") and isinstance(extnl.shape, tuple)):
            raise ValueError("Dependency external is not a Detector component.")
```